delta-app/src/main/java/io/cdap/delta/store/DefaultMacroEvaluator.java

```diff
@@ -43,11 +43,10 @@ public DefaultMacroEvaluator(Map<String, String> runtimeArgs, SecureStore secure
   @Override
   @Nullable
   public String lookup(String property) {
-    String val = runtimeArgs.get(property);
-    if (val == null) {
-      throw new InvalidMacroException(String.format("Argument '%s' is not defined.", property));
-    }
-    return val;
+    String msg = String.format("Macro cannot be resolved for property '%s' as macro resolution is not supported " +
```

Comment:
DefaultMacroEvaluator is also used below :

https://github.com/data-integrations/delta/blob/7ef527932d7ef4a9f830238a9d3e99916dd7e565/delta-app/src/main/java/io/cdap/delta/app/DeltaWorker.java#L182

I guess we only disable the use of Marco at Replicator design time (creating the replicator draft) but not at runtime (when the replicator is actually running) right ?

Reply:
I think its better to do it at runtime too for consistency. So if someone deploys pipeline bypassing the UI, it would result in the same error.